internal/state/graph/httproute.go

```diff
@@ -1,146 +1,359 @@
 package graph
 
 import (
+	"fmt"
+
 	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/validation/field"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/gateway-api/apis/v1beta1"
 
 	"github.com/nginxinc/nginx-kubernetes-gateway/internal/state/conditions"
+	"github.com/nginxinc/nginx-kubernetes-gateway/internal/state/validation"
 )
 
+// Rule represents a rule of an HTTPRoute.
+type Rule struct {
+	BackendGroup BackendGroup
+	// ValidMatches indicates whether the matches of the rule are valid.
+	// If the matches are invalid, NGK should not generate any configuration for the rule.
+	ValidMatches bool
+	// ValidFilters indicates whether the filters of the rule are valid.
+	// If the filters are invalid, the data-plane should return 500 error provided that the matches are valid.
+	ValidFilters bool
+}
+
+// ParentRef describes a reference to a parent in an HTTPRoute.
+type ParentRef struct {
+	// Gateway is the NamespacedName of the referenced Gateway
+	Gateway types.NamespacedName
+	// Idx is the index of the reference in the HTTPRoute.
+	Idx int
+}
+
 // Route represents an HTTPRoute.
 type Route struct {
 	// Source is the source resource of the Route.
 	// FIXME(pleshakov)
 	// For now, we assume that the source is only HTTPRoute.
 	// Later we can support more types - TLSRoute, TCPRoute and UDPRoute.
 	Source *v1beta1.HTTPRoute
+	// SectionNameRefs is a map of section names to the referenced NKG Gateways
+	SectionNameRefs map[string]ParentRef
+	// UnboundSectionNameRefs is a subset of SectionNameRefs that includes sections that could not be bound
+	// to the referenced Gateway. For example, because section does not exist in the Gateway.
+	UnboundSectionNameRefs map[string]conditions.Condition
```

Comment:
> Why store conditions as the value here? Are the conditions in this map not included in the Conditions field?

this condition is specific to a sectionRef and it is specific to attaching to that ref.  (1 ) There could be multiple refs pointing to an NKG gateway. (2) When we report status, we report status per each ref   --> so that's why we store it separately from the Conditions field

Reply:
https://github.com/nginxinc/nginx-kubernetes-gateway/pull/455/commits/cf2e5660a1e1baf922168dfda99bb3e89fa9a807